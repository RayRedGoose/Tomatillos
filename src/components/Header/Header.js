import React from 'react';
import './Header.scss';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const Header = ({ login }) => {
  let toggleLogin;
  login ?
  toggleLogin = <button id="logout" className="toggleLogin">Logout</button> : toggleLogin = <Link id="login-link" to="/login">
    <button id="login" className="toggleLogin">
      Login
    </button>
  </Link>;
  return (
    <header className="header">
      <h1>Rancid Tomatillos</h1>
      {toggleLogin}
    </header>
  )
}

export const mapStateToProps = state => ({
  login: state.login
});

export default connect(mapStateToProps)(Header);
