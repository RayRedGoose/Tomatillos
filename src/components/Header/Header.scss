@import '../../variables.scss';

.header {
  align-items: center;
  background-color: $main-background-color-transparent;
  border-bottom: 1px solid $main-accent-color;
  display: flex;
  height: 10vh;
  justify-content: space-between;
}

h1 {
  color: $titles-color;
  text-shadow: $text-shadow;
  font: {
    family: $large-text-font;
    size: 3rem;
  }
  margin-left: 1.5%;
}

.toggleLogin {
  background-color: $main-accent-color;
  border: none;
  border-radius: 45px;
  color: $main-text-color;
  cursor: pointer;
  font: {
    family: $small-text-font;
    size: 1rem;
    weight: 700;
  }
  height: 40px;
  outline: none;
  text-decoration: none;
  width: 100%;
  &:active {
    background-color: black;
    color: $main-text-color;
  }
}

#login-link {
  margin-right: 1.5%;
  width: 10%;
}

#logout {
  margin-right: 1.5%;
  width: 10%;
}

#sign-in-link {
  padding: -200px;
}
