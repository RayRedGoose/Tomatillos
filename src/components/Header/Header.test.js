import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';

describe('Header', () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(<Header login={false}/>);
    instance = wrapper.instance();
  });

  it('Should match the snapshot if login is false', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match the snapshot if login is true', () => {
    wrapper = shallow(<Header login={true}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('Should call the remove user method when the logout button is clicked', () => {
    wrapper = shallow(<Header user={true}/>);
    const mockDispatch = jest.fn();
    wrapper.find('#logout').simulate('click');
    expect(mockDispatch).toHaveBeenCalled();
  });

});
