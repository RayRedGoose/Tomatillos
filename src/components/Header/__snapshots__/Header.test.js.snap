// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match the snapshot if login is false 1`] = `
<header
  className="header"
>
  <h1>
    Rancid Tomatillos
  </h1>
  <Link
    id="login-link"
    to="/login"
  >
    <button
      className="toggleLogin"
      id="login"
    >
      Login
    </button>
  </Link>
</header>
`;

exports[`Header Should match the snapshot if login is true 1`] = `
<header
  className="header"
>
  <h1>
    Rancid Tomatillos
  </h1>
  <Link
    id="login-link"
    to="/login"
  >
    <button
      className="toggleLogin"
      id="login"
    >
      Login
    </button>
  </Link>
</header>
`;
